pkg/volume/csi/csi_block.go

```diff
@@ -171,8 +170,8 @@ func (m *csiBlockMapper) stageVolumeForBlock(
 	if csiSource.NodeStageSecretRef != nil {
 		nodeStageSecrets, err = getCredentialsFromSecret(m.k8s, csiSource.NodeStageSecretRef)
 		if err != nil {
-			return "", fmt.Errorf("failed to get NodeStageSecretRef %s/%s: %v",
-				csiSource.NodeStageSecretRef.Namespace, csiSource.NodeStageSecretRef.Name, err)
+			return "", volumetypes.NewTransientOperationFailure(log("failed to get NodeStageSecretRef %s/%s: %v",
```

Comment:
Are we able to add a test case for these changes?

Reply:
Okay, sorry re-reading the issue with full context. So this is for subequent calls to `MapVolume` after first one failed with a transient error. 

In general - this does mean that for any volume which didn't get mounted at all, must undergo some cleanup on pod deletion. IIRC, we do not have sufficient coverage for that code path, like what happens when you try to unmount a volume which was never mounted. 

I remember discussing this issue with @msau42 and we agreed that, for resizing issue, we could just ignore resizing errors during mount/map and solve this problem. Is there any reason, we are not doing that? I think, that will be a PR we can merge too.